src/Fmt_ast.ml

```diff
@@ -1744,91 +1744,22 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
-                        (break_unless_newline 1000 2)
-                    in
-                    let wrap_parens =
-                      wrap_k
-                        (fmt_or parens_bch
-                           (if imd then " (@;<1 2>" else " (@;<0 2>")
-                           "@;<1 2>")
-                        (fmt_if parens_bch (if imd then " )" else ")"))
-                    in
-                    hovbox 0
-                      ( ( match xcnd with
-                        | Some xcnd ->
-                            hvbox
-                              (if parens then -2 else 0)
-                              ( hvbox
-                                  (if parens then 0 else 2)
-                                  ( fmt_if (not first) "else "
-                                  $ fmt "if"
-                                  $ fmt_if_k first
-                                      (fmt_extension_suffix c ext)
-                                  $ fmt_attributes c ~pre:(fmt " ") ~key:"@"
-                                      pexp_attributes
-                                  $ fmt "@ " $ fmt_expression c xcnd )
-                              $ fmt "@ then" )
-                        | None -> fmt "else" )
-                      $ wrap_parens
-                          (fmt_expression c ~pro ~eol:(fmt "@;<1 2>")
-                             ~box:false ~parens:false xbch) )
-                    $ fmt_if (not last) "@ "
-                | `Keyword_first ->
-                    let wrap_parens =
-                      wrap_k
-                        (fmt_or parens_bch
-                           (if imd then " (@ " else " (@,")
-                           "@ ")
-                        (fmt_if parens_bch (if imd then " )" else ")"))
-                    in
-                    opt xcnd (fun xcnd ->
-                        hvbox 2
-                          ( fmt_or_k first
-                              (fmt "if" $ fmt_extension_suffix c ext)
-                              (fmt "else if")
-                          $ fmt_attributes c ~pre:(fmt " ") ~key:"@"
-                              pexp_attributes
-                          $ str " " $ fmt_expression c xcnd )
-                        $ fmt "@ " )
-                    $ hvbox 2
-                        ( fmt_or (Option.is_some xcnd) "then" "else"
-                        $ wrap_parens
-                            (fmt_expression c ~box:false ~parens:false xbch)
-                        )
-                    $ fmt_if (not last) "@ " )))
+                Params.get_if_then_else c.conf ~first ~last ~parens
+                  ~parens_bch ~xcond:xcnd ~expr_loc:pexp_loc
+                  ~fmt_extension_suffix:(fmt_extension_suffix c ext)
+                  ~fmt_attributes:
+                    (fmt_attributes c ~pre:(fmt " ") ~key:"@"
+                       pexp_attributes)
+                  ~fmt_cond:(fmt_expression c)
+                |> fun (p : Params.if_then_else) ->
```

Comment:
Why not `let p = .. in` ?

Reply:
I don't remember either why I wrote it like this for the other calls to `Params.*`.